pkg/storage/unified/resource/server.go

```diff
@@ -765,6 +769,10 @@ func (s *server) Origin(ctx context.Context, req *OriginRequest) (*OriginRespons
 
 // Index returns the search index. If the index is not initialized, it will be initialized.
 func (s *server) Index(ctx context.Context) (*Index, error) {
+	if err := s.Init(ctx); err != nil {
```

Comment:
Calling init will initialize the backend before creating the index. This is important as it will do things like run migrations before we start building the index.

Reply:
maybe we should add a comment